app/codes/committeemanager.py

```diff
@@ -116,3 +138,17 @@ def get_committee_for_current_block(last_block=None):
     committee = weighted_random_choices(miners, weights, committee_size)
     committee = sorted(committee, key=lambda d: d['wallet_address']) 
     return committee
+
+
+def is_miner_committee_cached(last_block_hash):
+    global miner_committee_cache
+    timestamp = get_corrected_time_ms()
+
+    if (miner_committee_cache['current_block_hash'] == last_block_hash
+        and miner_committee_cache['timestamp'] < timestamp + BLOCK_TIME_INTERVAL_SECONDS * 1000):
```

Comment:
@kousthubraja this should be closer to 1 second because in some instances (like invalid block by valid miner) the negative consensus may arrive faster than block_interval_time.

Reply:
this line of code is for performance to not calculate committee and miner for same block multiple times. there was a bug in this line found now.  should be `timestamp - BLOCK_TIME_INTERVAL_SECONDS ` instead of `timestamp + BLOCK_TIME_INTERVAL_SECONDS`